src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +27,26 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) {
+            return true;
+        }
+        if (object == null) {
+            return false;
+        }
+        Rectangle rectangle = (Rectangle) object;
+        return this.width.equals(rectangle.width) && this.length.equals(rectangle.length)
+                && this.color.equals(rectangle.color);
+    }
+
+    @Override
+    public int hashCode() {
+        int hash = this.width * this.length;
```

Comment:
what will happen if width or length ==0?
try do smth like this 
https://www.baeldung.com/java-hashcode